### ISSUE TITLE
Remove deprecated LoopRunner API

### ISSUE THREAD

**Issue by fjetter** [dask/distributed#7239]
We deprecated functionality around LoopRunner that allowed to pass closed/not-running loops to it, see https://github.com/dask/distributed/issues/6680

It's been ~3 months now and we should remove the deprecated functionality. This also relates to tornado 6.2 and py3.11 compat, see https://github.com/dask/distributed/pull/7134

Closes https://github.com/dask/distributed/issues/6163

@wence- you started looking into this. Are you still interested?

cc @graingert

**Comment by wence-** [comment]
> @wence- you started looking into this. Are you still interested?

Yes. I got some way in but then because quite confused about whose responsibility it _should_ be to start the event loop. Removing the final `make_current` in `Nanny._run` wasn't sufficient to remove usage of deprecated asyncio functionality AFAICT.
